### PR TITLE
Models

### PR DESCRIPTION
**Pull request by nancylee713**
## What's this PR do?
Modify `models.py`, add additional arguments (i.e., `nullable`, `unique`) and relationship between tables

### Where should the reviewer start?
Review `models.py`

### How should this be manually tested?
1. Make sure that you are working in a virtual environment (check for (venv)).
2. Run the python interpreter, by typing `python3` in the terminal.
3. Type the following inside the interpreter:

```python
# Import files and create a db
>>> from app import db
>>> from app import User, UserProfile
>>> db.create_all()

# Add a user to User model
>>> user_1 = User(email='test@email.com', password='password')
>>> db.session.add(user_1)
>>> db.session.commit()
>>> user_1
User(id: '5', email: 'test@email.com')


# Add a user_profile to UserProfile model
>>> user_1_profile = UserProfile(name='nancy', age=28, proficiency='beginner', user_id=5)
>>> db.session.add(user_1_profile)
>>> db.session.commit()


# Check relationship between user and user_profile table
>>> user_1.profile
[UserProfile(name: 'nancy', level: 'beginner', user_id: '5')]

>>> user_1_profile.user
User(id: '5', email: 'test@email.com')
```

### What are the relevant tickets?
#14 

#### Screenshots (if appropriate)

- Note: other tables that were created in the previous PR should be modified similarly.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,9 @@ venv
 __pycache__
 
 .vscode
+
+migrations
+
+google_cloud_sdk.json
+
+.gitignore
```
Comment (sschipke) [inline .gitignore]: I always forget to add the `.gitignore` file. Nice job!

File: app.py [hunk 2]
```diff
@@ -44,7 +44,14 @@ def get_users_profile():
...
-      
+
+@app.route('/speech_questions')
+def get_speech_questions():
+    try:
+        users = SpeechQuestion.query.all()
+        return jsonify([e.serialize() for e in users])
+    except Exception as e:
+            return(str(e))
+
```
Comment (sschipke) [inline app.py]: Do you think we'll need to get speech questions based on difficulty, as opposed to all of them at once?

File: models.py [hunk 3]
```diff
@@ -32,26 +40,25 @@ class UserProfile(db.Model):
...
   id = db.Column(db.Integer, primary_key=True, autoincrement=True)
-  name = db.Column(db.String())
+  name = db.Column(db.String(),unique=True, nullable=False)
   age = db.Column(db.Integer)
-  proficiency = db.Column(db.String())
-  user_id = db.Column(db.Integer, ForeignKey(User.id))
-  created_date = db.Column(DateTime, default=datetime.datetime.utcnow)
-  updated_date = db.Column(DateTime, default=datetime.datetime.utcnow, onupdate=datetime.datetime.utcnow)
+  proficiency = db.Column(db.String(), nullable=False)
+  user_id = db.Column(db.Integer, ForeignKey(User.id), nullable=False)
```
Comment (sschipke) [inline models.py]: Good idea making this not `nullable`

File: app.py (hunk 2 above)
Comment (nancylee713) [reply app.py]: Yes, there should be methods in sqlalchemy that allow us to filter speech questions based on level of difficulty

**Comment by nancylee713** [comment]
There are 3 models (i.e., GrammarQuestion, UserSpeech, and UserGrammar) that are left untouched. I will modify them similar to the SpeechQuestion model.

**Pull request merged**